### PR TITLE
SDIT-1983: ✨ Add sentence start date and primary sentence

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/OffenceHistoryDetailDto.kt [hunk 1]
```diff
@@ -35,5 +35,7 @@ data class OffenceHistoryDetailDto(
     this.courtDate,
     this.caseId,
     null,
+    null,
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/OffenceHistoryDetailDto.kt:38]: this is when mapped outside of JPA, so by setting to `null` the fields won't be returned i.e. no change to other endpoints

Comment (petergphillips) [reply src/main/java/uk/gov/justice/hmpps/prison/api/model/OffenceHistoryDetailDto.kt:38]: although I guess they will still be in the swagger docs

File: src/test/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderChargeTransformerTest.kt [hunk 2]
```diff
@@ -68,4 +70,57 @@ class OffenderChargeTransformerTest {
...
+  fun `should set sentence information to null if no associated sentences`() {
+    val offenderHistoryDetails = OffenderCharge.builder()
+      .offenderBooking(OffenderBooking.builder().bookingId(2L).build())
+      .offenderCourtCase(OffenderCourtCase.builder().id(1L).build())
+      .offence(Offence.builder().code("code").description("desc").statute(Statute.builder().code("st code").build()).build())
+      .build().let {
+        OffenderChargeTransformer().convert(it)
+      }
+    assertThat(offenderHistoryDetails.sentenceStartDate).isNull()
+    assertThat(offenderHistoryDetails.primarySentence as Boolean?).isNull()
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderChargeTransformerTest.kt:84]: had to use `as Boolean?` here to stop a null pointer as it tried to convert from the java to kotlin `Boolean`

File: src/main/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderChargeTransformer.kt [hunk 3]
```diff
@@ -27,6 +28,8 @@ class OffenderChargeTransformer : Converter<OffenderCharge, OffenceHistoryDetail
       .secondaryResultConviction(offenderCharge.resultCodeTwo?.isConvictionFlag)
       .courtDate(latestCourtEvent?.eventDate)
       .offenceSeverityRanking(offenderCharge.offence.severityRanking?.toInt() ?: Int.MAX_VALUE)
+      .sentenceStartDate(sentence?.sentenceStartDate)
+      .primarySentence(if (sentence != null) sentence.consecutiveToSentenceSequence == null else null)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderChargeTransformer.kt:32]: only set to `true` or `false` if there is a sentence

File: src/main/resources/db/migration/data/R__9_10_OFFENDER_SENTENCE_CHARGES.sql [hunk 4]
```diff
@@ -9,3 +9,7 @@ VALUES(-20, 2, -9, TO_DATE('2021-05-25', 'YYYY-MM-DD'), 'USER');
 INSERT INTO OFFENDER_SENTENCE_CHARGES
 (OFFENDER_BOOK_ID, SENTENCE_SEQ, OFFENDER_CHARGE_ID, CREATE_DATETIME, CREATE_USER_ID)
 VALUES(-20, 3, -8, TO_DATE('2021-05-25', 'YYYY-MM-DD'), 'USER');
+
+INSERT INTO OFFENDER_SENTENCE_CHARGES
+(OFFENDER_BOOK_ID, SENTENCE_SEQ, OFFENDER_CHARGE_ID, CREATE_DATETIME, CREATE_USER_ID)
+VALUES(-20, 4, -15, TO_DATE('2021-05-25', 'YYYY-MM-DD'), 'USER');
```
Comment (petergphillips) [inline src/main/resources/db/migration/data/R__9_10_OFFENDER_SENTENCE_CHARGES.sql:15]: linked to an existing `OFFENDER_SENTENCE` record

File: src/main/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderChargeTransformer.kt (hunk 3 above)
Comment (mikehalmamoj) [reply src/main/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderChargeTransformer.kt:32]: I'd normally use `?` for this, e.g. `sentence?.let { it.consecutuveToSequence == null }`

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/OffenceHistoryDetailDto.kt (hunk 1 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/hmpps/prison/api/model/OffenceHistoryDetailDto.kt:38]: I think that is fine, I have done that and just added a note to swagger to explain

File: src/test/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderChargeTransformerTest.kt (hunk 2 above)
Comment (andymarke) [reply src/test/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderChargeTransformerTest.kt:84]: Nice; I wondered what the fix for that was

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**